### ISSUE TITLE
Does not build for Visual Studio 2022

### ISSUE THREAD

**Issue by PeterFromSweden** [log4cplus/log4cplus#550]
Does not build for Visual Studio 2022, 10756 errors
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2039	'basic_string_view': is not a member of 'std'	log4cplus	 ...source\repos\log4cplus\include\log4cplus\tstring.h	42	
Version 52dd360e04d0fcdadc87cc1874884d9bbc027f99
OS: Win11
Visual Studio 2022 (17.1.6)
Build of solution after retarget to new Visual Studio version

**Comment by wilx** [comment]
I do not have access to Visual Studio on Windows any more. Do you have any suggestion for a fix. As far as I can see, the `tstring.h` looks correct. It does include the `<string_view>` header.

https://github.com/log4cplus/log4cplus/blob/52dd360e04d0fcdadc87cc1874884d9bbc027f99/include/log4cplus/tstring.h#L35

**Comment by PeterFromSweden** [comment]
I found out that log4cpp is a well maintained nuget package which is much
easier to use in visual studio.
/Peter

Den ons 4 maj 2022 16:55Václav Haisman ***@***.***> skrev:

**Comment by StianWAa** [comment]
You'll have to change the C++ Language Standard to C++17 in the *.props files.
Eg. the file **log4cplus.props** should get a:  
`<LanguageStandard>stdcpp17</LanguageStandard>`  
inside its `<ClCompile>` section.

**Comment by PeterFromSweden** [comment]
Ok. Improved a lot, but not all.

Severity	Code	Description	Project	File	Line	Suppression State
Error	C1083	Cannot open include file: 'QtGlobal': No such file or directory	Qt5DebugAppender	<...>\log4cplus\qt5debugappender\qt5debugappender.cxx	38	
Error	C1083	Cannot open include file: 'QtGlobal': No such file or directory	Qt4DebugAppender	<...>\log4cplus\qt4debugappender\qt4debugappender.cxx	38	
Error	C2969	syntax error: '=': expected member function definition to end with '}'	CLFSAppender	<...>\log4cplus\src\clfsappender.cxx	75	
Error	C4430	missing type specifier - int assumed. Note: C++ does not support default-int	CLFSAppender	<...>\log4cplus\src\clfsappender.cxx	75	
Error	C2612	trailing '=' illegal in base/member initializer list	CLFSAppender	<...>\log4cplus\src\clfsappender.cxx	75	
Error	C1004	unexpected end-of-file found	CLFSAppender	<...>\log4cplus\src\clfsappender.cxx	303	
Error	C2969	syntax error: '=': expected member function definition to end with '}'	MSTTSAppender	<...>\log4cplus\src\msttsappender.cxx	179	
Error	C4430	missing type specifier - int assumed. Note: C++ does not support default-int	MSTTSAppender	<...>\log4cplus\src\msttsappender.cxx	179	
Error	C2612	trailing '=' illegal in base/member initializer list	MSTTSAppender	<...>\log4cplus\src\msttsappender.cxx	179	
Error	C1004	unexpected end-of-file found	MSTTSAppender	<...>\log4cplus\src\msttsappender.cxx	341	
Error	C1083	Cannot open source file: '..\..\tests\customloglevel_test\customloglevel.cxx': No such file or directory	customloglevel_testS	<...>\log4cplus\msvc17\tests\c1xx	1

**Comment by wilx** [comment]
Please try building with CMake instead. That should work.

**Comment by rabindra-harlalka** [comment]
@PeterFromSweden , I faced the same problem as you did. The solution for me was to not clone the repository but to download one of the release packages (e.g. https://github.com/log4cplus/log4cplus/releases/download/REL_2_0_8/log4cplus-2.0.8.zip).

The reason I believe, apart from the C++17 standard pointed above, is that when we clone the repository, the git client doesn't clone the submodules catch and threadpool. Perhaps there may be a way to instruct git to clone the submodules as well, but I found that just downloading the released source package is much easier. Interestingly, the latter one doesn't even require us to play with project setting or props file w.r.t. C++17 support.

Perhaps you already figured it out, but I am leaving the comment here in case anyone finds it useful.
